baseplate/__init__.py

```diff
@@ -147,13 +147,13 @@ def from_upstream(
         :raises: :py:exc:`ValueError` if any of the values are inappropriate.
 
         """
-        if trace_id is None or not 0 <= trace_id < 2 ** 64:
+        if trace_id is None:
             raise ValueError("invalid trace_id")
 
-        if span_id is None or not 0 <= span_id < 2 ** 64:
+        if span_id is None:
             raise ValueError("invalid span_id")
 
-        if parent_id is None or not 0 <= parent_id < 2 ** 64:
+        if parent_id is None:
             raise ValueError("invalid parent_id")
```

Comment:
hmm I believe this part of implementation is actually slightly different from the spec.

the spec says services should pass `"0"` as the header value to explicitly state that the given id does not exist, and then the server should just generate a new one randomly (for trace id) or just treat itself's server span as a root span (for parent/span id). we might want to check against empty string and "0" here and generate random ones instead?

Reply:
I don't see anything about `"0"` in the spec. Do you mean this?

> If valid trace headers are present on the request, they MUST be used to configure the span object. If no headers are present, or they are invalid, implementations MUST generate a new random 64-bit integer for Trace ID, set span ID to the same value, and leave parent ID as null.

That happens when the exception gets raised: 
1. https://github.com/reddit/baseplate.py/blob/develop/baseplate/frameworks/thrift/__init__.py#L91-L99
2. https://github.com/reddit/baseplate.py/blob/develop/baseplate/__init__.py#L470-L471

Reply:
I could have sworn I've seen `0` mentioned in the spec, maybe that's on an older version of the spec, or maybe an older version of the implementation of basepalte.py, or maybe my memory just failed me 😅

Reply:
Oh now I remembered, this line in current code will fallback to use "0" (instead of empty string) when serializing to zipkin json and that's likely where I get that impression from: https://github.com/reddit/baseplate.py/blob/673fe736ac004a2f141ae7b25c6c47ec7af7edef/baseplate/observers/tracing.py#L282

Reply:
ah, funky!